### PR TITLE
Add Ingress spec for Dashboard

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Add Ingress Spec For Dashboard and add manager to sync dashboard ingress.  

Note that the the dashboard ingress and TlsCluster couldn't be both enabled currently as the Ingress assumes the TLS termination. FYI: https://kubernetes.io/docs/concepts/services-networking/ingress/#tls 

### Check List 

 - E2E test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support Ingress for Dashboard in TidbCluster Spec
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -1161,3 +1166,29 @@ type RestoreStatus struct {
 	TimeCompleted metav1.Time        `json:"timeCompleted"`
 	Conditions    []RestoreCondition `json:"conditions"`
 }
+
+// +k8s:openapi-gen=true
+// DashBoard describe the TiDB Dashboard desired state
+type DashBoardSpec struct {
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/types.go:1172]: `DashBoardSpec` -> `DashboardSpec`

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -1161,3 +1166,29 @@ type RestoreStatus struct {
...
+	// PDDashboardIngress describe the ingress of the TiDB Dashboard
+	// +optional
+	Ingress *IngressSpec `json:"ingress,omitempty"`
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/types.go:1175]: it is possible to use Kubernetes builtin ingress types directly?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/types.go:1175]: I suggest not as we should control the path that the ingress would only expose the dashboard api excluding pdapi.

File: pkg/apis/pingcap/v1alpha1/types.go (hunk 1 above)
Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/types.go]: updated.

**Pull request closed without merging**